src/components/DSL/NewFlow.tsx

```diff
@@ -32,6 +32,7 @@ export const NewFlow: FunctionComponent<PropsWithChildren> = () => {
          * we don't need to do anything special, just add a new flow if
          * supported
          */
+        setSettings({ dsl, namespace });
```

Comment:
We need to update the current DSL whenever we add a new flow, since in this code path when [there are no flows, it assumes that the DSL is the same](https://github.com/KaotoIO/kaoto-ui/pull/2115/files#diff-a16b0f9b3604fc7b64098ed9eb1ff564b5333d8a6a7656a2f650c245d60144d1R27), because `Array.every` returns `true` for empty arrays.

```
[].every(() => true);
// prints `true`
```